auth-server/src/main/java/com/antonio/authserver/service/GroupService.java

```diff
@@ -56,37 +63,37 @@ public void deleteGroupByName(String name) {
             groupRepository.deleteByName(name);
         } else {
             throw new CustomException(
-                    "Group with name [ " + groupRepository.findByName(name) + "does not exist", HttpStatus.BAD_REQUEST
+                    "Group with name [ " + name + "does not exist", HttpStatus.BAD_REQUEST
             );
         }
     }
 
     public GroupDto findGroupByName(String name) {
         Optional<UserGroup> byGroupName = groupRepository.findByName(name);
+
         if (byGroupName.isPresent()) {
             return groupMapperClass.daoToDto(byGroupName.get());
         } else {
             throw new CustomException(
-                    "Group with the name [ " + byGroupName.get().getName() + " ] does not exists!",
+                    "Group with the name [ " + name + " ] does not exists!",
                     HttpStatus.CONFLICT);
         }
     }
 
     public void updateByName(String name, GroupDto group) {
-        Optional<UserGroup> oldGroup = groupRepository.findByName(name);
-
-        if (oldGroup.isPresent()) {
-            if (!group.getName().isEmpty() || !group.getName().equals(" ")) {
-                oldGroup.get().setName(group.getName());
-            }
-            if (!group.getUsers().isEmpty() || group.getUsers().size() > 0) {
-                oldGroup.get().setAppUserGroup(group.getUsers());
-            }
+       UserGroup oldGroup = groupRepository.findByName(name).orElseThrow(() -> new CustomException("Group with name " + name + " not found", HttpStatus.BAD_REQUEST));
+       UserGroup newGroup = groupMapperClass.dtoToDao(group);
```

Comment:
Why do you need newGroup?

Reply:
I needed it for making it DAO, because I have to return a AppUser list not DTO List..